adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m

```diff
@@ -88,79 +88,101 @@ - (void)renderNativeAdForAdConfiguration:
                  strongSelf->_nativeAd = nativeAd;
                  strongSelf->_nativeAd.delegate = strongSelf;
                  strongSelf->_nativeAd.rootViewController = adConfiguration.topViewController;
-
-                 if (strongSelf->_loadCompletionHandler) {
-                   id<GADMediationNativeAdEventDelegate> delegate =
-                       strongSelf->_loadCompletionHandler(strongSelf, nil);
-                   strongSelf->_delegate = delegate;
-                 }
+      
+                 [strongSelf loadRequiredData];
                }];
 }
 
-#pragma mark - GADMediationNativeAd
-
-- (GADNativeAdImage *)icon {
-  if (!_icon) {
-    if (_nativeAd.data.icon && _nativeAd.data.icon.imageURL != nil) {
-      _icon = [self imageWithUrlString:_nativeAd.data.icon.imageURL];
+- (void)loadRequiredData {
+    GADPangleRTBNativeRenderer *__weak weakSelf = self;
+    void (^localBlock)(void) = ^{
+        GADPangleRTBNativeRenderer *strongSelf = weakSelf;
+        if (strongSelf && strongSelf->_loadCompletionHandler) {
+            strongSelf->_delegate = strongSelf->_loadCompletionHandler(strongSelf,nil);
+        }
+    };
+    NSString *URLString = _nativeAd.data.icon.imageURL;
+    if (!URLString.length) {
+        localBlock();
+        return;
     }
-  }
+    NSURL *url = [NSURL URLWithString:URLString];
+    if (!url) {
+        localBlock();
+        return;
+    }
+    NSURLSession *session = [NSURLSession sharedSession];
+    NSURLSessionDataTask *task = [session dataTaskWithURL:url completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if (!self) {
```

Comment:
I assume that this is intended to be another `weakSelf` / `strongSelf` construct. `self` will always be `nonnull` here because this is retaining it.

Reply:
ok~